open-ce/docker_build.py

```diff
@@ -130,13 +133,45 @@ def build_in_container(image_name, output_folder, arg_strings):
     # Cleanup
     _stop_container(container_name)
 
-def build_with_docker(output_folder, arg_strings):
+
+
+def _generate_dockerfile_name(build_types, cuda_version):
+    '''
+    Ensure we have valid combinations.  I.e. Specify a valid cuda version
+    '''
+    if 'cuda' in build_types:
+        dockerfile = os.path.join(BUILD_CUDA_IMAGE_PATH, "Dockerfile.cuda-" + cuda_version)
+        build_image_path = BUILD_CUDA_IMAGE_PATH
+        if not os.path.isfile(dockerfile):
+            raise OpenCEError(Error.UNSUPPORTED_CUDA, cuda_version)
+    else:
+        #Build with cpu based image
+        dockerfile = os.path.join(BUILD_IMAGE_PATH, "Dockerfile")
+        build_image_path = BUILD_IMAGE_PATH
+    return build_image_path, dockerfile
+    
+def _capable_of_cuda_containers(cuda_versions):
+    '''
+    Check if we can run containers with Cuda installed.  This can be accomplished in two ways
+    First if the host server does not have a driver installed
+    Second, if the host driver is compatible with the level of cuda being used in the image
+    '''
+    
+    return True if not utils.cuda_driver_installed() or utils.cuda_level_supported(cuda_versions) else False
```

Comment:
```suggestion
    return not utils.cuda_driver_installed() or utils.cuda_level_supported(cuda_versions)
```

No need to have the if statement